src/Common/tests/System/Net/Http/Http2Frames.cs

```diff
@@ -0,0 +1,286 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using System.Net.Sockets;
+using System.Security.Authentication;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Test.Common
+{
+    
+    public enum FrameType : byte
+    {
+        Data = 0,
+        Headers = 1,
+        Priority = 2,
+        RstStream = 3,
+        Settings = 4,
+        PushPromise = 5,
+        Ping = 6,
+        GoAway = 7,
+        WindowUpdate = 8,
+        Continuation = 9,
+
+        Last = 9
+    }
+
+    [Flags]
+    public enum FrameFlags : byte
+    {
+        None = 0,
+
+        // Some frame types define bits differently.  Define them all here for simplicity.
+        EndStream =     0b00000001,
+        Ack =           0b00000001,
+        EndHeaders =    0b00000100,
+        Padded =        0b00001000,
+        Priority =      0b00100000,
+
+        ValidBits =     0b00101101
+    }
+
+    public class Frame
```

Comment:
This frame class is largely copied from the actual implementation in SocketsHttpHandler. I made the call not to share the code because I needed to remove all of the validation in order to test using malformed frames.